CONTRIBUTING.md

```diff
@@ -0,0 +1,256 @@
+Then return to your GitHub home page, find your "percona-server" repository and clone it locally.
+Note
+that [ creating SSH keys and putting public one to the GitHub ](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account)
+settings will save a lot of time for you with GitHub workflow.
+
+### Before you start coding 
+
+In the cloned local repository checkout the "trunk" branch you are going to work on. For 8.0.X releases is 8.0. For
+8.4.X series it is 8.4, and for 9.0.X it is "trunk".
+
+Now you are ready to create a branch for your contribution. Please name the branch according to the following pattern:
+
+```
+PS-9876-8.0-bug_in_some_module
+^^^^^^^
+Jira issue number
+        ^^^
+        Base version
+            ^^^^^^^^^^^^^^^^^^^
+            Very short definition of the issue 
+``` 
+
+The next step is to compile the code in your new branch before making changes.
+At his point, you need to fetch third-party code which is organized as Git sub-modules. Go to the
+project directory and type the following:
+
+```
+git submodule update --init
+``` 
+
+The Git will fetch required third-party modules into the source tree.
+
+### First build of the source code
+
+Now we can try to build the project. Below is a quick and dirty shell script for that (works with GCC 14). It gives you
+hints how to do outside-the-tree build and how to set CMake flags.
+
+```
+#!/bin/bash
+
+CMAKE_FLAGS="-DCMAKE_BUILD_TYPE=Debug \
+ -DDOWNLOAD_BOOST=1 \
+ -DWITH_BOOST=/home/${USER}/projects/boost\
+ -DWITH_DEBUG=ON\
+ -DWITH_ROCKSDB=ON\
+ -DWITH_COREDUMPER=OFF\
+ -DWITH_CURL=system\
+ -DWITH_FIDO=system\
+ -DWITH_ZLIB=bundled\
+ -DCMAKE_CXX_FLAGS=-Wno-error=template-id-cdtor -Wno-error=dangling-reference \
+ -Wno-error=deprecated-declarations
+ 
+SRC_DIR=$1
+if [ -z ${SRC_DIR} ] ; then
+  echo "Please provide source directory as 1st parameter!"
+  exit 1
+fi
+if [ -d $SRC_DIR ] ; then
+  echo "Source dir is $SRC_DIR"
+else
+  echo  "Source dir $SRC_DIR does not exist"
+  exit 1
+fi
+ABS_SRC_DIR=`realpath $SRC_DIR`
+if [ -z $2 ] ; then
+  BUILD_DIR="build-${SRC_DIR}"
+else
+  BUILD_DIR=$2
+fi
+echo "Build dir is $BUILD_DIR"
+if [ -d $BUILD_DIR ] ; then
+ echo "Build dir exists. Cmake will re-use cached data." 
+else
+ mkdir $BUILD_DIR
+fi
+cd $BUILD_DIR
+cmake $CMAKE_FLAGS $ABS_SRC_DIR
+make -j 16
+
+```
+
+Compiling from a source requires patience. MySQL has many dependencies on different libraries. The good hint is to install build dependencies for the "upstream" version of MySQL. 
+For example, on Ubuntu ( don't forget to enable sources in _software-properties-gtk ) run following command:
+```
+apt build-dep mysql-server-core-8.0
+```
+
+Another hint is to check [ documentation on  compilation ](https://docs.percona.com/percona-server/8.0/compile-percona-server.html) .
+
+If it does not help, we recommend a try and fail approach to installing the development packages.
```

Comment:
Why would the user persist in compiling the code?

Reply:
First, not the user, but the developer. Second, successful compilation is the absolutely mandatory step. This is how the developer gets running executable with his changes in the code